### PR TITLE
docs: add snapshot-and-restore.md

### PR DESCRIPTION
**Pull request by ushitora-anqou**
Add additional documentation for the Snapshot&Restore functionality. Derived from https://github.com/topolvm/topolvm/issues/827

### CODE REVIEW

**Review by daichimukai — COMMENTED** [review]
Nice!

**Comment by toshipp** [comment]
Please update README.md
https://github.com/topolvm/topolvm/blob/main/docs/README.md

IMO, it should be in the getting started section.

**Comment by toshipp** [comment]
also, this could be deleted.
https://github.com/topolvm/topolvm/blob/main/docs/advanced-setup.md#snapshot

File: docs/snapshot-and-restore.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+```sh
+$ kubectl exec -it my-pod -- bash
+root@my-pod:/# echo hello > /data/world
+root@my-pod:/# ls /data/world
+/data/world
+root@my-pod:/# cat /data/world
+hello
+```
+
+### Take a Snapshot
```
[the remark this answers is not in the document]
Comment (ushitora-anqou) [reply docs/snapshot-and-restore.md:104]: I updated my patch. Could you review it again? The changes are: https://github.com/topolvm/topolvm/compare/debae299312f8eba2cdbf55fac646628dbd7860d..e3a6e95ffedfae6c1e5372398532f7e604262dd8

**Review by llamerada-jp — APPROVED** [review]

File: docs/snapshot-and-restore.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (daichimukai) [reply docs/snapshot-and-restore.md:104]: LGTM. Thank you.

**Review by daichimukai — APPROVED** [review]

**Pull request merged**